core/src/main/java/bisq/core/xmr/connection/MoneroConnectionManager.java

```diff
@@ -0,0 +1,144 @@
+package bisq.core.xmr.connection;
+
+import bisq.core.xmr.connection.model.MoneroConnection;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.time.Duration;
+
+import java.net.URI;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+
+import monero.common.MoneroRpcConnection;
+
+@Slf4j
+@Singleton
+public class MoneroConnectionManager {
+
+    private final Object lock = new Object();
+
+    private final monero.common.MoneroConnectionManager connectionManager;
+
+    @Inject
+    public MoneroConnectionManager(monero.common.MoneroConnectionManager connectionManager) {
```

Comment:
Remove package qualification after this class renamed to `MoneroConnectionsManager`.